merge_streams_block.py

```diff
@@ -0,0 +1,54 @@
+from nio.common.block.attribute import Input
+from nio.common.block.base import Block
+from nio.common.discovery import Discoverable, DiscoverableType
+from nio.common.signal.base import Signal
+from nio.metadata.properties import VersionProperty, TimeDeltaProperty, \
+    BoolProperty
+from nio.modules.scheduler import Job
+
+
+@Input('input_2')
+@Input('input_1')
+@Discoverable(DiscoverableType.block)
+class MergeStreams(Block):
+
+    """ Take two input streams and combine signals together. """
+
+    expiration = TimeDeltaProperty(default={})
+    notify_once = BoolProperty(default=True)
+    version = VersionProperty('0.1.0')
+
+    def __init__(self):
+        super().__init__()
+        self._signals = { "input_1": {}, "input_2": {} }
+        self._expiration_jobs = { "input_1": None, "input_2": None }
+
+    def process_signals(self, signals, input_id='default'):
+        merged_signals = []
+        for signal in signals:
+            self._signals[input_id] = signal
+            if self._signals["input_1"] and self._signals["input_2"]:
+                merged_signals.append(self._merge_signals())
+        if self.expiration:
+            self._schedule_signal_expiration_job(input_id)
+        if merged_signals:
+            self.notify_signals(merged_signals)
+
+    def _merge_signals(self):
+        """ Merge signals 1 and 2 and clear from memory if only notify once """
+        merged_signal = self._signals["input_1"].to_dict()
```

Comment:
My guess is that 1 is causing something with 2. Something about setting `__class__` or `__dict__` on the new signal probably messes up the Python internals. Question 1 is definitely a bug and should be logged in nio. I do think hidden attributes should be merged though.

Maybe as a temporary workaround, until 1 is fixed, something like this:
```python
sig_1_dict = self._signals["input_1"].to_dict(hidden=True)
sig_2_dict = self._signals["input_2"].to_dict(hidden=True)
# Remove all double underscored keys from the dictionary
# until hidden signal bug is fixed
for key in sig_1_dict.keys():
    if key.startswith('__'):
        del sig_1_dict[key]
for key in sig_2_dict.keys():
    if key.startswith('__'):
        del sig_2_dict[key]
out_sig = {}
out_sig.update(sig_1_dict)
out_sig.update(sig_2_dict)
```
This prevents the block from needing to be updated later on to support hidden properties, but also allows it to use hidden properties in the mean-time until the bug is fixed.